SwissKnife/avro/AvroTransformer.py

```diff
@@ -0,0 +1,271 @@
+        return record_with_casted_values
+
+    def get_renamed_record_and_remove_invalid_fields(self, record: Record) -> Record:
+        """Transforms a record to another one with correct names: If a field of the record
+        uses an alias, it will be changed to the correct name using the avro schema as reference.
+        Also, a field that is not in the avro schema will be removed.
+        :param record: The input record.
+        :type record: Record
+        :return: A record with the correct field names and without invalid fields.
+        :rtype: Record
+        """
+        new_record = {}
+        for key, value in record.items():
+            # Remove not existing keys
+            if key not in self.rename_dict:
+                continue
+            new_key = self.rename_dict[key]
+            new_record[new_key] = value
+        return new_record
+
+    def get_record_with_defaults(self, record: Record) -> Record:
+        """Transforms a record no a new one where empty field will
+           be filled with its default value.
+           If a field is empty and has not default value,
+           the function will raise an exception.
+        :param record: The input record.
+        :type record: Record
+        :raises RuntimeError: When a field empty and has not default value.
+        :return: A new record with default values applied.
+        :rtype: Record
+        """
+        new_record = {}
+        for key, defaults_value in self.defaults_dict.items():
+            if key not in record and defaults_value is NoDefault:
+                raise RuntimeError(f"Required  {key} not in record {record}")
+            elif key not in record and defaults_value is not NoDefault:
+                new_record[key] = defaults_value
+            else:
+                new_record[key] = record[key]
+        return new_record
+
+    def get_transformed_record(self, record: Record) -> Record:
+        """Applies the transformations in a record using the avro schema
+           as a reference.
+        :param record: The input record.
+        :type record: Record
+        :return: A new record with transformations applied.
+        :rtype: Record
+        """
+        new_record = {}
+        for key, value in record.items():
+            if key in self.transform_dict:
+                transform_function = self.transform_dict[key]
+                new_record[key] = transform_function(value, record)
+            else:
+                new_record[key] = value
+        return new_record
+
+    @staticmethod
+    def _create_rename_dict(schema: dict) -> dict:
+        """Creates a dict that maps a name with the correct one.
+        :param schema: The provided schema.
+        :type schema: dict
+        :return: A dict with the map name -> correct_name. It can has
+        a "name -> name" pair or an "alias -> name" pair.
+        :rtype: dict
+        """
+
+        rename_dict = {}
+        for field in schema["fields"]:
```

Comment:
It seems like it might be a good idea to move this strings which match the keys of the schema ("fields", "name", "aliases", ....) into a class variable. What do you think?

Reply:
Good idea, very convenient!